pallets/treasury/src/lib.rs

```diff
@@ -1,51 +1,54 @@
 #![cfg_attr(not(feature = "std"), no_std)]
 #![recursion_limit = "256"]
 
-use pallet_balances as balances;
 use pallet_identity as identity;
 use polymesh_common_utilities::{
-    traits::{
-        balances::Trait as BalancesTrait, identity::Trait as IdentityTrait, CommonTrait,
-        NegativeImbalance, PositiveImbalance,
-    },
+    traits::{balances::Trait as BalancesTrait, identity::Trait as IdentityTrait, CommonTrait},
     Context,
 };
-use polymesh_primitives::{AccountKey, Beneficiary, IdentityId};
+use polymesh_primitives::{traits::IdentityCurrency, AccountKey, Beneficiary, IdentityId};
 
 use codec::Encode;
 use frame_support::{
-    debug, decl_error, decl_event, decl_module, decl_storage,
+    decl_error, decl_event, decl_module,
     dispatch::DispatchResult,
     ensure,
     traits::{Currency, ExistenceRequirement, Imbalance, OnUnbalanced, WithdrawReason},
 };
 use frame_system::{self as system, ensure_root, ensure_signed};
-use sp_runtime::traits::Saturating;
+use sp_runtime::{
+    traits::{AccountIdConversion, Saturating},
+    ModuleId,
+};
 use sp_std::{convert::TryFrom, prelude::*};
 
 pub type ProposalIndex = u32;
 
 type Identity<T> = identity::Module<T>;
+type BalanceOf<T> =
+    <<T as Trait>::Currency as Currency<<T as frame_system::Trait>::AccountId>>::Balance;
+type NegativeImbalanceOf<T> =
+    <<T as Trait>::Currency as Currency<<T as frame_system::Trait>::AccountId>>::NegativeImbalance;
+
+/// The treasury's module id, used for deriving its sovereign account ID.
+const MODULE_ID: ModuleId = ModuleId(*b"py/trsry");
```

Comment:
We could move it to a common module in order to share this constant between `Treasury` and `Identity` pallets.